--- cheops-ng-0.2.3/Makefile.in~	2011-05-25 23:01:39.385455243 +0200
+++ cheops-ng-0.2.3/Makefile.in	2011-05-25 23:01:41.218432328 +0200
@@ -5,6 +5,7 @@
 DEBUG_FLAGS=# -DDEBUG_DISCOVER -DDEBUG_EVENT -DDEBUG_IO -DDEBUG_SCHEDULER
 
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
 
 CC=@CC@
@@ -75,7 +76,6 @@ TEST_OBJS = $(NORMOBJS) agent-io.o logge
 
 TESTS = test-sched test-socket test-probe
 COMLIBS = #@COMLIBS@
-CC_LIBS = 
 CFLAGS += -Wall $(DEBUG_FLAGS) $(GTKCFLAGS)  $(XMLCFLAGS)
 
 ifdef DEBUG
@@ -92,8 +92,8 @@ all: version.h installdir.h $(PROGS)
 
 static: version.h $(COMOBJS) $(GUIOBJS) $(GTKOBJS) $(COMLIBS) $(AGENTOBJS) $(AGENTLIBS) $(COMOBJS) $(NORMOBJS) $(COMLIBS)
 	rm cheops-ng cheops-agent -f
-	$(CC) -o cheops-ng $(COMOBJS) $(GUIOBJS) $(GTKOBJS) $(COMLIBS) $(GTKLIBS) $(CC_LIBS) $(STATIC_FLAGS) -static 
-	$(CC) -o cheops-agent $(COMOBJS) $(AGENTOBJS) $(COMLIBS) $(AGENTLIBS) $(NORMOBJS) $(CC_LIBS) $(AGENTLIBFLAGS) $(GLIB) $(STATIC_FLAGS) -static
+	$(CC) -o cheops-ng $(LDFLAGS) $(COMOBJS) $(GUIOBJS) $(GTKOBJS) $(COMLIBS) $(GTKLIBS) $(STATIC_FLAGS) -static 
+	$(CC) -o cheops-agent $(LDFLAGS) $(COMOBJS) $(AGENTOBJS) $(COMLIBS) $(AGENTLIBS) $(NORMOBJS) $(AGENTLIBFLAGS) $(GLIB) $(STATIC_FLAGS) -static
 
 strip: cheops-ng cheops-agent
 	strip cheops-ng
@@ -115,13 +115,13 @@ version.h: version
 	echo "#endif" >> version.h
 
 cheops-ng: version.h $(COMOBJS) $(GUIOBJS) $(GTKOBJS) $(COMLIBS)
-	$(CC) -o cheops-ng $(COMOBJS) $(GUIOBJS) $(GTKOBJS) $(COMLIBS) $(GTKLIBS) $(CC_LIBS) $(STATIC_FLAGS)
+	$(CC) -o cheops-ng $(LDFLAGS) $(COMOBJS) $(GUIOBJS) $(GTKOBJS) $(COMLIBS) $(GTKLIBS) $(STATIC_FLAGS)
 
 cheops-agent: version.h $(AGENTOBJS) $(AGENTLIBS) $(COMOBJS) $(NORMOBJS) $(COMLIBS)
-	$(CC) -o cheops-agent $(COMOBJS) $(AGENTOBJS) $(COMLIBS) $(AGENTLIBS) $(NORMOBJS) $(CC_LIBS) $(AGENTLIBFLAGS) $(GLIB) $(STATIC_FLAGS)
+	$(CC) -o cheops-agent $(LDFLAGS) $(COMOBJS) $(AGENTOBJS) $(COMLIBS) $(AGENTLIBS) $(NORMOBJS) $(AGENTLIBFLAGS) $(GLIB) $(STATIC_FLAGS)
 
 cheops-sh: $(COMOBJS) $(SHOBJS) $(NORMOBJS) $(COMLIBS)
-	$(CC) -o cheops-sh $(COMOBJS) $(SHOBJS) $(COMLIBS) $(SHLIBS) $(NORMOBJS) $(CC_LIBS) $(SHLIBFLAGS)
+	$(CC) -o cheops-sh $(LDFLAGS) $(COMOBJS) $(SHOBJS) $(COMLIBS) $(SHLIBS) $(NORMOBJS) $(SHLIBFLAGS)
 
 test-sched: $(TEST_OBJS) test-sched.o
 	$(CC) -o test-sched $(TEST_OBJS) $(COMLIBS) test-sched.o
@@ -142,13 +142,13 @@ gpl.c: COPYING
 	( echo "char *gpl_text = \"\\" ; cat COPYING | sed s/\\\"/\\\\\"/g | awk -v ORS="\\n" '{print $$0"\\" }'; echo "\";" ) >gpl.c
 
 agent-io.o: io.c
-	$(CC) $(CFLAGS) -c io.c -o agent-io.o $(CC_LIBS)
+	$(CC) $(CFLAGS) -c io.c -o agent-io.o $(LDFLAGS)
 
 agent-event.o: event.c
-	$(CC) $(CFLAGS) -c event.c -o agent-event.o $(CC_LIBS)
+	$(CC) $(CFLAGS) -c event.c -o agent-event.o $(LDFLAGS)
 
 agent-ip_utils.o: ip_utils.c
-	$(CC) $(CFLAGS) -c ip_utils.c -o agent-ip_utils.o $(CC_LIBS) $(GLIB)
+	$(CC) $(CFLAGS) -c ip_utils.c -o agent-ip_utils.o $(LDFLAGS) $(GLIB)
 
 dep:
 	makedepend *.c $(CFLAGS) $(GTKCFLAGS) -I./
